teps/0128-scheduled-runs.md

```diff
@@ -0,0 +1,393 @@
+---
+status: implementable
+title: Scheduled Runs
+creation-date: '2022-12-20'
+last-updated: '2022-12-20'
+authors:
+- '@vdemeester'
+- '@sm43'
+- '@lbernick'
+collaborators: []
+---
+
+# TEP-0128: Scheduled Runs
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+- [Proposal](#proposal)
+  - [Bindings](#bindings)
+  - [CLI](#cli)
+- [Design Evaluation](#design-evaluation)
+  - [Reusability](#reusability)
+  - [Simplicity](#simplicity)
+  - [Flexibility](#flexibility)
+  - [Conformance](#conformance)
+  - [Performance](#performance)
+  - [Security](#security)
+  - [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+  - [CronTrigger CRD with fixed event body](#crontrigger-crd-with-fixed-event-body)
+  - [Add schedule to Trigger](#add-schedule-to-trigger)
+  - [Add schedule to EventListener](#add-schedule-to-eventlistener)
+  - [Create a PingSource](#create-a-pingsource)
+- [Implementation Plan](#implementation-plan)
+- [Implementation Pull Requests](#implementation-pull-requests)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+This TEP proposes allowing Task/Pipeline users to schedule Task/Pipeline runs on a recurring basis.
+
+## Motivation
+
+Currently, to get a run to occur on a regular basis, a user must write a cron job that either creates the run itself
+(which duplicates much of triggers' functionality) or pings an EventListener.
+If the EventListener is exposed outside of the cluster, the user must also sign the cron job event body and verify it in a custom interceptor
+(although it's unlikely a user would choose to expose an ingress for an EventListener used only for cron jobs).
+
+### Goals
+
+- Can create new PipelineRuns, TaskRuns, or CustomRuns on a recurring basis without creating a webhook or writing a CronJob
+
+### Non-Goals
+
+- Polling a repository (or other external system) for changes; this is covered in [TEP-0083](./0083-polling-runs-in-tekton.md).
+
+### Use Cases
+
+- Release Pipeline: User would like to create releases of their project on a nightly basis.
+
+## Proposal
+
+Create a new CronTrigger CRD that fires on a specified schedule.
+For example:
+
+```yaml
+apiVersion: triggers.tekton.dev/v1alpha1
+kind: CronTrigger
+metadata:
+  name: nightly-release
+spec:
+  schedule: "0 0 * * *"
+  triggers:
+  - triggerRef: experimental-release-pipeline
```

Comment:
Do we have plan to support reference for Eventlistener as well ?

Thinking like  CronTrigger will trigger based on given schedule so if ref is for Eventlistener then it will create Eventlistener object

Reply:
To check my understanding, you're wondering if we'd plan to support a reference to an EventListener within a CronTrigger, and the CronTrigger would send a fixed event to that EventListener on a schedule? I don't have plans to include that in a proposal but I'm curious if there's a use case you have in mind for that feature.

Reply:
No usecase as of now but just thought like if someone need to execute EL based on cronTrigger

But ya proposal can be enhanced in future if we get such use case scenario